index.md

```diff
@@ -7,5 +7,44 @@ Best practices are difficult to practice without a good environment in which
 to practice them. Here we enumerate the elements of a productive development
 environment.
 
-_Note that this is currently a placeholder; we will be adding content
+##<a name="environments"></a> Environments
+When developing and maintaining software several environments will be needed. Usually at least 4 environments are needed, each servering a different purpose:
+
+* Dev (development)
+    * Where developers have free rein(Admin) to do whatever they need to for development, such as install software for evaluation, test new architecures, or test production install instructions. 
+    * Code changes from Dev will most likely create new Int builds.
+    * Dev should be "sandboxed" from other environments to prevent any accidents to other resources.
+* Int (integration)
+    * Where developers combine code from everyone and test to make sure they merged correctly, and it works as expected. 
+    * Builds that pass in Int can be passed to Test for possible use in Prod.
+* Test
+    * Permissions and data are exact copies from Prod. 
+    * The data is updated on some predefined schedule determined by communication between developers and system admins. 
+    * End users and other testers verify bug fixes and new features. 
+    * Any build that passes in Test is eligible for promotion to Prod.
+* Prod (production)
+    * Where the built software is deployed for use by all users. 
+    * Actual data is entered, and software provides value. 
+    * Developers to have as little admin access as needed. 
+    * Prod also has a well defined process for deploying new test builds that have been promoted. See [Build Promotion] (#build-promotion) below.
+
+It should be noted that most (if not all) environments will need the full set of developer tools, this includes but is not limited to:
+
+* Compilier(s) / build tool(s)
```

Comment:
s/Compilier/Compiler/ – but I'm included to remove the list and end this sentence at the comma.

I think it's reasonable to expect the full development toolchain to be available on all servers but that starts getting into deployment practices and the desirability of avoiding in-band builds for deployment. That's not a necessity but it definitely makes testing and deployment faster and more reliable.